pkgs/desktops/budgie/plugins/budgie-extras/mkBudgieExtrasPlugin.nix

```diff
@@ -0,0 +1,114 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, substituteAll
+, budgie
+, sound-theme-freedesktop
+, util-linux
+, vorbis-tools
+, xdg-utils
+, bash
+, libnotify
+, locale
+, plank
+, procps
+, wmctrl
+, xdotool
+, gnome
+, xorg
+, glib
+, intltool
+, meson
+, ninja
+, pkg-config
+, python3
+, vala
+}:
+
+{
+  # Name of the directory the plugin is located at.
+  pluginName
+  # Name of the plugin in the meson_options.txt file.
+, mesonOptionName ? pluginName
+  # Name of the plugin entrypoint, found under the "Module" key in the .plugin
+  # file. Only used by Python plugins.
+, moduleName ? pluginName
+  # Is this a Python plugin?
+, isPython ? false
+, ...
+} @ args:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "budgie-${pluginName}";
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "UbuntuBudgie";
+    repo = "budgie-extras";
+    rev = "v${finalAttrs.version}";
+    fetchSubmodules = true;
+    hash = "sha256-juPUs5qtY0Sqzh+AxuuWwJL34mL4t/bTbTFmfNg2JrY=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./fix-paths.patch;
+      # FIXME: This points to the patch outPath, not the plugin's
```

Comment:
In evolution-data-server we do https://github.com/NixOS/nixpkgs/blob/ea37b82acd5737fa9f250f1452afafc3903ba4f7/pkgs/desktops/gnome/core/evolution-data-server/default.nix#L74-L77